### ISSUE TITLE
allOf generates invalid code

### ISSUE THREAD

**Issue by mwlazlo** [anymaniax/orval#251]
## What are the steps to reproduce this issue?
1. Create query-objects.yaml:
```yaml
openapi: 3.0.0
info:
  title: test
  version: 0.0.1
paths:
  /test:
    get:
      responses:
        "200":
          description: Test
          content:
            application/json:
              schema:
                $ref: "models.yaml#/components/schemas/CustomResponse"
```
2. Create models.yaml:
```yaml
openapi: 3.0.0
info:
  title: Shared Models
  version: 1.0.0
paths: {}
components:
  schemas:
    BasicResponse:
      type: object
      title: BasicResponse
      properties:
        next:
          type: string
        prev:
          type: string
    CustomResponse:
      type: object
      allOf:
        - $ref: '#/components/schemas/BasicResponse'
        - type: object
          properties:
            data:
              type: array
              items:
                type: string
          required:
            - data
```
3. Create build.js
```js
module.exports = {
  spense: {
    input: './query-objects.yaml',
    output: {
      mode: 'single',
      target: 'query-objects.ts',
      client: 'react-query',
      mock: true,
    },
  },
};

```
4. Run Orval
`orval --config build.js`
## What happens?
Orval generates invalid TypeScript:
```typescript
query-objects.ts:71:39 - error TS1136: Property assignment expected.

71 export const getGetTestMock = () => ({{},data: [...Array(faker.datatype.number({min: 1, max: 10}))].map(() => (faker.random.word()))})
                                         ~
```

**Comment by anymaniax** [comment]
Thanks for reporting the issue. I will look into it when I have a bit of time

**Comment by anymaniax** [comment]
Should be fixed with the next release

**Comment by anymaniax** [comment]
Should be fixed with `6.4.0` can you test when you have time, please?

**Comment by anymaniax** [comment]
Closing this issue since it's normally fixed. Don't hesitate if you have any problem
